### PR TITLE
add addons section to tutorial

### PR DESCRIPTION
**Pull request by toddjordan**
Part of issue #1374 

Here's the start to the addons section.  Moved in mirage, and added a new section on our custom style addon.

### CODE REVIEW

File: source/localizable/tutorial/installing-addons.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
+Ember has a rich ecosystem of addons that can be easily added to projects.
+Addons provide a wide range of functionality to projects, often saving time and letting you focus on your project's value.
```
Comment (acorncom) [inline source/localizable/tutorial/installing-addons.md]: focus on your project

File: source/localizable/tutorial/installing-addons.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+To browse addons, visit the [Ember Observer](https://emberobserver.com/) website.  It catalogs and categorizes ember addons that have been published to NPM and assigns them a score based on a variety of criteria.
+
+For Super Rentals, we'll take advantage of two addons: [ember-cli-tutorial-style](https://github.com/toddjordan/ember-cli-tutorial-style) and [ember-cli-mirage](http://www.ember-cli-mirage.com/).
```
Comment (acorncom) [inline source/localizable/tutorial/installing-addons.md]: Should we update this style link now?

File: source/localizable/tutorial/installing-addons.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+### ember-cli-tutorial-style
+
+Instead of having you copy/paste in CSS to style Super Rentals, we've created an addon called [ember-cli-tutorial-style](https://github.com/toddjordan/ember-cli-tutorial-style) that instantly adds CSS to the tutorial.
+The addon works by creating a file called `ember-tutorial.css` and putting that file in the application's `/vendor` directory.
```
Comment (acorncom) [inline source/localizable/tutorial/installing-addons.md]: the **super-rentals** vendor directory?

File: source/localizable/tutorial/installing-addons.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+At compile time the contents of the css file is automatically concatenated to the `vendor.css` file which is imported into the application's `apps/index.html`.
+By creating the file in the `vendor` directory, the addon is exposing it to the developer for potential customization (in case we want to tweak the styles).
```
Comment (acorncom) [inline source/localizable/tutorial/installing-addons.md]: I think we can skip some of this explanatory text. Maybe something along these lines instead?

As Ember-CLI runs, it takes the addon's CSS file and puts it in `vendor.css` (which is referenced in `/app/index.html`). We can make additional style tweaks in `/app/styles/app.css`

File: source/localizable/tutorial/installing-addons.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+Run the following command to install the addon:
+
+```shell
+ember install ember-cli-tutorial-style
+```
+
+Starting the server will incorporate the new css and refreshing the screen will give you this:
```
Comment (acorncom) [inline source/localizable/tutorial/installing-addons.md]: refreshing **our browser window** will give us this

File: source/localizable/tutorial/installing-addons.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+![super rentals styled homepage](../../images/installing-addons/styled-super-rentals-basic.png)
+
+### ember-cli-mirage
+
+[Mirage](http://www.ember-cli-mirage.com/) is a client HTTP stubbing library often used for Ember acceptance testing.
+For the case of this tutorial, we'll use mirage as our data source.
```
Comment (acorncom) [inline source/localizable/tutorial/installing-addons.md]: For this tutorial, we'll use Mirage as our source of data

File: source/localizable/tutorial/installing-addons.md [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+
+Install the Mirage addon as follows:
+
+```shell
+ember install ember-cli-mirage
+```
+
+If you were running `ember serve` in another shell, restart the server to include Mirage in your build.
+
+Let's now configure Mirage to send back our rentals that we had defined above by updating `app/mirage/config.js`:
```
Comment (acorncom) [inline source/localizable/tutorial/installing-addons.md]: Sidenote: Mirage is moving all files to `/mirage/` in v0.2.0, so we'll need to update the tutorial when v0.2.0 releases

File: source/localizable/tutorial/routes-and-templates.md [hunk 8]
```diff
@@ -105,22 +109,21 @@ Since we will be using the `contact` template, the `contact` route does not need
...
+  <p>Ember Rentals Representatives would love to help you<br>choose a destination or answer
+    any questions you may have.</p>
+  <p>
+    Ember Rentals HQ
+    <address>
+      1212 Test Address Avenue<br>
+      Testington, OR 97233
+    </address>
+    <a href="tel:503.555.1212">+1 (503) 555-1212</a><br>
+    <a href="mailto:emberrentalsrep@emberrentals.com">superrentalsrep@superrentals.com</a>
```
Comment (acorncom) [inline source/localizable/tutorial/routes-and-templates.md]: Are we Ember Rentals or Super Rentals? This line aptly illustrates the mixed message we're sending above ;-)

File: source/localizable/tutorial/service.md [hunk 9]
```diff
@@ -299,5 +303,8 @@ export default Ember.Object.extend({
   }
 
 });
-
 ```
+
+You should now see some end to end maps functionality show up on our front page!
```
Comment (acorncom) [inline source/localizable/tutorial/service.md]: We should

File: source/localizable/tutorial/simple-component.md [hunk 10]
```diff
@@ -21,131 +21,173 @@ installing component-test
   create tests/integration/components/rental-listing-test.js
 ```
 
+We'll start by implementing a failing test with the image toggling behavior that we want.
+
+For our integration test, we'll create a stub rental that has all the properties that our rental model has.
+We will assert that the component is initially rendered without the `wide` class name. and that clicking on the image will add the class `wide` to our element, and clicking it a second time with take the `wide` class away.
```
Comment (acorncom) [inline source/localizable/tutorial/simple-component.md]: class name. Clicking the image will add

File: source/localizable/tutorial/simple-component.md [hunk 11]
```diff
@@ -21,131 +21,173 @@ installing component-test
...
+    We hope you find exactly what you're looking for in a place to stay.
+    <br>Browse our listings, or use the search box above to narrow your search.
+  </p>
+  {{#link-to 'about' class="button"}}
+    About Us
+  {{/link-to}}
+</div>
 
 {{#each model as |rentalUnit|}}
   {{rental-listing rental=rentalUnit}}
```
Comment (acorncom) [inline source/localizable/tutorial/simple-component.md]: Should we have a minus section showing what we took out here?

File: source/localizable/tutorial/simple-component.md [hunk 12]
```diff
@@ -21,131 +21,173 @@ installing component-test
...
-<p>Type: {{rental.type}}</p>
-<p>Location: {{rental.city}}</p>
-<p>Number of bedrooms: {{rental.bedrooms}}</p>
-{{#if isImageShowing}}
-  <p><img src={{rental.image}} alt={{rental.type}} width="500"></p>
-  <button {{action "imageHide"}}>Hide image</button>
-{{else}}
-  <button {{action "imageShow"}}>Show image</button>
-{{/if}}
+To make it where clicking on the image link shows the image to the user,
```
Comment (acorncom) [inline source/localizable/tutorial/simple-component.md]: To allow the user to widen the image

File: source/localizable/tutorial/simple-component.md [hunk 13]
```diff
@@ -21,131 +21,173 @@ installing component-test
...
+  <div class="detail">
+    <span>Number of bedrooms:</span> {{rental.bedrooms}}
+  </div>
+</article>
 ```
 
-Then let's setup an `imageHide` action handler in our component to set `isImageShowing` to `false`:
+Clicking the anchor element will send the action to the component.
+Ember will then go into the `actions` hash and call the `toggleImageSize` function.
+Let's create the `toggleImageSize` function and toggle `isWide` property on our component:
```
Comment (acorncom) [inline source/localizable/tutorial/simple-component.md]: toggle **the** `isWide`

File: source/localizable/tutorial/simple-component.md [hunk 14]
```diff
@@ -21,131 +21,173 @@ installing component-test
...
-      this.set('isImageShowing', false);
+    toggleImageSize() {
+      this.toggleProperty('isWide');
     }
   }
 });
 ```
 
-Now our users can toggle images on and off using the "Show image" and "Hide image" buttons.
+Now when we click the image or the `View Larger` link in our browser, we can see our image larger,
```
Comment (acorncom) [inline source/localizable/tutorial/simple-component.md]: browser, we see our image show larger,

**Comment by acorncom** [comment]
LGTM :+1:

File: source/localizable/tutorial/routes-and-templates.md (hunk 8 above)
Comment (locks) [inline source/localizable/tutorial/routes-and-templates.md]: Do we own that domain? Are we deploying the app? I don't like having email links to whoever knows :\

File: source/localizable/tutorial/installing-addons.md (hunk 7 above)
Comment (locks) [inline source/localizable/tutorial/installing-addons.md]: Can we ship using the 0.2 betas? It's the way forward and has better support for json:api as far as I could tell.

File: source/localizable/tutorial/installing-addons.md (hunk 4 above)
Comment (locks) [inline source/localizable/tutorial/installing-addons.md]: Ember CLI

**Comment by toddjordan** [comment]
Thanks for the comments @acorncom !

Looking to incorporate feedback tonight

File: source/localizable/tutorial/installing-addons.md (hunk 4 above)
Comment (toddjordan) [inline source/localizable/tutorial/installing-addons.md]: I went with slightly different wording that I thought was more accurate, but still tried to simplify as you did.

File: source/localizable/tutorial/installing-addons.md (hunk 7 above)
Comment (toddjordan) [inline source/localizable/tutorial/installing-addons.md]: we are shipping using the 0.2 beta and its working fine.  Its a typo above.

File: source/localizable/tutorial/routes-and-templates.md (hunk 8 above)
Comment (toddjordan) [inline source/localizable/tutorial/routes-and-templates.md]: true.  we can make the email domain something we do own, like ember weekly;-P

File: source/localizable/tutorial/simple-component.md (hunk 11 above)
Comment (toddjordan) [inline source/localizable/tutorial/simple-component.md]: yep, do now

**Comment by toddjordan** [comment]
@acorncom @locks updated pr based on feedback

File: source/localizable/tutorial/autocomplete-component.md [hunk 15]
```diff
@@ -250,3 +254,6 @@ export default function() {
 }
 ```
 
+After updating your mirage config, you should see passing tests, as well as a simple filter on your home screen, that will update the rental list as you type:
```
Comment (acorncom) [inline source/localizable/tutorial/autocomplete-component.md]: We generally speak in a "we" or plural form, but this sentence shifts the other way.  Let's update so it reads more like:

> After updating our Mirage config, we should see passing tests

File: source/localizable/tutorial/installing-addons.md [hunk 16]
```diff
@@ -0,0 +1,84 @@
...
+
+To browse addons, visit the [Ember Observer](https://emberobserver.com/) website.  It catalogs and categorizes ember addons that have been published to NPM and assigns them a score based on a variety of criteria.
+
+For Super Rentals, we'll take advantage of two addons: [ember-cli-tutorial-style](https://github.com/toddjordan/ember-cli-tutorial-style) and [ember-cli-mirage](http://www.ember-cli-mirage.com/).
+
+### ember-cli-tutorial-style
+
+Instead of having you copy/paste in CSS to style Super Rentals, we've created an addon called [ember-cli-tutorial-style](https://github.com/ember-learn/ember-cli-tutorial-style) that instantly adds CSS to the tutorial.
+The addon works by creating a file called `ember-tutorial.css` and putting that file in the super-rentals `vendor` directory.
+As Ember CLI runs, it takes the `ember-tutorial` CSS file and puts it in `vendor.css` (which is referenced in `/app/index.html`).We can make additional style tweaks to `/vender/ember-tutorial.css`, and the changes will take effect whenever we restart the app.
```
Comment (acorncom) [inline source/localizable/tutorial/installing-addons.md]: Nice.  We're missing a space between the period and We in "the .We can make" part of this line

File: source/localizable/tutorial/installing-addons.md [hunk 17]
```diff
@@ -0,0 +1,84 @@
...
+
+Run the following command to install the addon:
+
+```shell
+ember install ember-cli-tutorial-style
+```
+
+Starting the server will incorporate the new css and refreshing our browser window will give you this:
```
Comment (acorncom) [inline source/localizable/tutorial/installing-addons.md]: CSS

**Comment by acorncom** [comment]
@toddjordan a couple more tweaks here.  Let's merge once we've tweaked those

**Comment by toddjordan** [comment]
@acorncom tweaks updated!

**Comment by acorncom** [comment]
@homu r+

**Pull request merged**